File: registry/layers/framework/go-net-http/files/handlers.go
package app

import (
	"encoding/json"
	"net/http"
	"os"
)

type healthPayload struct {
	Status   string `json:"status"`
	Service  string `json:"service"`
	Database string `json:"database"`
	SDK      string `json:"sdk"`
}

func NewMux() *http.ServeMux {
	mux := http.NewServeMux()
	mux.HandleFunc("/health", func(writer http.ResponseWriter, request *http.Request) {
		writer.Header().Set("Content-Type", "application/json; charset=utf-8")
		_ = json.NewEncoder(writer).Encode(healthPayload{
			Status:   "ok",
			Service:  "{{serviceName}}",
			Database: "{{databaseProvider}}",
			SDK:      os.Getenv("SDK_PROVIDER"),
		})
	})
	return mux
}

File: registry/layers/framework/deno-edge/files/handlers.ts
type Handler = (request: Request) => Response | Promise<Response>

interface HealthPayload {
  status: 'ok'
  service: string
  database: string
  sdk: string
  runtime: 'deno'
}

function health(): Response {
  const payload: HealthPayload = {
    status: 'ok',
    service: '{{serviceName}}',
    database: '{{databaseProvider}}',
    sdk: Deno.env.get('SDK_PROVIDER') ?? 'none',
    runtime: 'deno',
  }
  return Response.json(payload)
}

export const routes: Record<string, Handler> = {
  '/health': health,
}
